### PR TITLE
Issue 65: replace lodash with native/sublibs

### PR DESCRIPTION
**Pull request by mikeguta**
A solution for issue #65 (remove lodash as a dependency)
I chose to retain 3 of the `lodash` sublibs (`lodash.merge`, `lodash.capitalize` and `lodash.startcase`) as PROD dependencies, just because they are non-trivial.

### CODE REVIEW

File: __tests__/decorators.test.ts [hunk 1]
```diff
@@ -219,7 +218,9 @@ describe('decorators', () => {
       }
     }
 
-    routes = _.keyBy(parseRoutes(getMetadataArgsStorage()), 'action.method')
+    parseRoutes(getMetadataArgsStorage()).forEach(
+      (route) => (routes[route.action.method] = route)
+    )
```
Comment (epiphone) [inline __tests__/decorators.test.ts:223]: nit: could use `.reduce` here to match previous behaviour

File: src/generateSpec.ts [hunk 2]
```diff
@@ -88,14 +97,13 @@ export function getHeaderParams(route: IRoute): oa.ParameterObject[] {
...
-    .value()
 
-  const headersMeta = _.find(route.params, { type: 'headers' })
+  const headersMeta = route.params.find((p) => p.type === 'headers')
   if (headersMeta) {
     const schema = getParamSchema(headersMeta) as oa.ReferenceObject
     headers.push({
       in: 'header',
-      name: _.last(_.split(schema.$ref, '/')) || '',
+      name: schema.$ref.split('/').slice(-1)[0] || '',
```
Comment (epiphone) [inline src/generateSpec.ts:106]:
```suggestion
      name: schema.$ref.split('/').pop() || '',
```
this works too for getting the last element

File: src/generateSpec.ts [hunk 3]
```diff
@@ -158,12 +168,12 @@ export function getQueryParams(
...
     })
-    .value()
 
-  const queriesMeta = _.find(route.params, { type: 'queries' })
+  const queriesMeta = route.params.find((p) => p.type === 'queries')
   if (queriesMeta) {
     const paramSchema = getParamSchema(queriesMeta) as oa.ReferenceObject
-    const paramSchemaName = _.last(_.split(paramSchema.$ref, '/')) || ''
+    // the last segment after '/'
+    const paramSchemaName = paramSchema.$ref.split('/').slice(-1)[0] || ''
```
Comment (epiphone) [inline src/generateSpec.ts:176]:
```suggestion
    const paramSchemaName = paramSchema.$ref.split('/').pop() || ''
```
the same here

**Review by epiphone — CHANGES_REQUESTED** [review]
Looks great! Let's get the CI errors fixed before merging

File: __tests__/decorators.test.ts (hunk 1 above)
Comment (mikeguta) [reply __tests__/decorators.test.ts]: `reduce` instead of `forEach` 75495ff

File: src/generateSpec.ts (hunk 2 above)
Comment (mikeguta) [reply src/generateSpec.ts]: `pop` used here 7993cc4

File: src/generateSpec.ts (hunk 3 above)
Comment (mikeguta) [reply src/generateSpec.ts]: `pop` used here 7993cc4

**Comment by mikeguta** [comment]
Hi @epiphone sorry for the delay, not had much spare time lately. I think this is now sorted, please do the honours when you have a bit of time.

**Review by epiphone — APPROVED** [review]
Thanks!

**Pull request merged**